packages/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts

```diff
@@ -73,127 +137,45 @@ const useBoxStyles = makeStyles({
-    borderWidth: theme.global.strokeWidth.thin,
-    ':hover': {
-      borderColor: theme.alias.color.neutral.neutralStrokeAccessibleHover,
-    },
-    ':active': {
-      borderColor: theme.alias.color.neutral.neutralStrokeAccessiblePressed,
-    },
-  }),
-
-  checked: theme => ({
-    backgroundColor: theme.alias.color.neutral.compoundBrandBackground,
-    borderWidth: 0,
-    ':hover': {
-      backgroundColor: theme.alias.color.neutral.compoundBrandBackgroundHover,
-    },
-    ':active': {
-      backgroundColor: theme.alias.color.neutral.compoundBrandBackgroundPressed,
-    },
-  }),
+const useInputStyles = makeStyles({
+  input: {
+    opacity: 0,
+    position: 'absolute',
+    margin: 0,
+    padding: 0,
+    cursor: 'pointer',
+  },
 
-  mixed: theme => ({
-    borderColor: theme.alias.color.neutral.compoundBrandStroke,
-    borderWidth: theme.global.strokeWidth.thin,
-    ':hover': {
-      borderColor: theme.alias.color.neutral.compoundBrandStrokeHover,
-    },
-    ':active': {
-      borderColor: theme.alias.color.neutral.compoundBrandStrokePressed,
-    },
-  }),
+  disabled: {
+    cursor: 'default',
+  },
 });
 
 const useIndicatorStyles = makeStyles({
-  indicator: {
-    position: 'absolute',
+  box: theme => ({
+    width: '100%',
+    height: '100%',
+    fill: 'currentColor',
+    overflow: 'hidden',
     display: 'flex',
     alignItems: 'center',
     justifyContent: 'center',
-    fill: 'currentColor',
-  },
-
-  // TODO: Remove fontSize once checkbox uses react-icons
-  medium: {
-    fontSize: '8px',
-    width: '8px',
-    height: '8px',
-  },
-
-  // TODO: Remove fontSize once checkbox uses react-icons
-  large: {
-    fontSize: '10px',
-    width: '10px',
-    height: '10px',
-  },
-
-  disabled: theme => ({
-    opacity: 1,
-    color: theme.alias.color.neutral.neutralForegroundDisabled,
-  }),
-
-  unchecked: {
-    opacity: 0,
-  },
-
-  // TODO: neutralForegroundInverted change to NeutralForegroundOnBrand once it's added
-  checked: theme => ({
-    opacity: 1,
-    color: theme.alias.color.neutral.neutralForegroundInverted,
+    position: 'absolute',
+    boxSizing: 'border-box',
+    borderStyle: 'solid',
+    borderRadius: theme.global.borderRadius.small,
   }),
 
-  mixed: theme => ({
-    opacity: 1,
-    color: theme.alias.color.neutral.compoundBrandForeground1,
-    ':hover': {
-      color: theme.alias.color.neutral.compoundBrandForeground1Hover,
-    },
-    ':active': {
-      color: theme.alias.color.neutral.compoundBrandForeground1Pressed,
-    },
+  circular: theme => ({
+    borderRadius: theme.global.borderRadius.circular,
   }),
-});
-
-const useLabelStyles = makeStyles({
-  label: {
-    userSelect: 'none',
-    cursor: 'pointer',
-  },
 
   disabled: theme => ({
+    borderWidth: theme.global.strokeWidth.thin,
+    borderColor: theme.alias.color.neutral.neutralStrokeDisabled,
     color: theme.alias.color.neutral.neutralForegroundDisabled,
```

Comment:
I think this will cause the checkmark to show up on disabled+unchecked checkboxes. Perhaps instead of using `color: 'transparent'` above to hide the checkmark, you could add a style that sets the children to opacity 0:
```ts
  unchecked: {
    '& > *': {
      opacity: 0,
    },
  },
```

That'll hide the checkmark for unchecked checkmarks, regardless of what `color` is set to.